src/goto-instrument/contracts/utils.cpp

```diff
@@ -163,3 +167,67 @@ void disable_pointer_checks(source_locationt &source_location)
   source_location.add_pragma("disable:pointer-primitive-check");
   source_location.add_pragma("disable:pointer-overflow-check");
 }
+
+void simplify_gotos(goto_programt &goto_program, namespacet &ns)
+{
+  for(auto &instruction : goto_program.instructions)
+  {
+    if(instruction.is_goto())
+    {
+      const auto &condition = instruction.get_condition();
+      const auto &simplified = simplify_expr(condition, ns);
+      if(simplified.is_false())
+        instruction.turn_into_skip();
+    }
+  }
+}
```

Comment:
This PR also depends on https://github.com/diffblue/cbmc/pull/6527, perhaps it doesn't have to?

Reply:
This one supersedes [the](https://github.com/diffblue/cbmc/pull/6527), which has been closed.